LibraBFT/ImplShared/Consensus/Types.agda

```diff
@@ -162,6 +202,66 @@ module LibraBFT.ImplShared.Consensus.Types where
   α-EC-RM : RoundManager → EpochConfig
   α-EC-RM rm = α-EC ((_rmEC rm) , (_rmEC-correct rm))
 
+
+  record GoodLens {A} (l : Lens RoundManagerEC A) : Set where
+    field
+      getRM : ∀ (rm : RoundManager) (f : A → A)
+            → Σ RoundManager (λ rm' → _rmEC rm' ≡ over l f (_rmEC rm))
+
+  -- These are the parts of the RoundManager that depend on an
+  -- EpochConfig. We do not particularly care which EpochConfig
+  -- they care about yet.
+  --
+  lBlockTree : Lens RoundManagerEC BlockTree
+  lBlockTree = rmBlockStore ∙ bsInner
+
+
+  rmSyncOnly-manual : Lens RoundManagerEC Bool
+  rmSyncOnly-manual = mkLens' g s
+    where
+    g : RoundManagerEC → Bool
+    g rmec = _rmSyncOnly rmec
+
+    s : RoundManagerEC → Bool → RoundManagerEC
+    s rmec b = record rmec { _rmSyncOnly = b}
+
+  instance
+     -- Note that this is an autogenerated lens that doesn't come anywhere near anything to do with
+     -- epoch change, so is a good test example, though we can always define the top-level lenses
+     -- manually if this doesn't work.  Maybe that's where I should start.
```

Comment:
On this point, I think that a pseudo-composition operator is needed for`GoodLens`. Specifically, composing a `GoodLens` with a (compatible) `Lens` produces a new `GoodLens`.